chia/wallet/wallet_node.py

```diff
@@ -965,6 +965,11 @@ async def new_peak_wallet(self, new_peak: wallet_protocol.NewPeakWallet, peer: W
                 if far_behind or len(self.synced_peers) == 0:
                     syncing = True
                     self.wallet_state_manager.set_sync_mode(True)
+
+                if not syncing and not (self._secondary_peer_sync_task is None or self._secondary_peer_sync_task.done()):
```

Comment:
this should be 
"if  (self._secondary_peer_sync_task is None or self._secondary_peer_sync_task.done()):"

"syncing" is True if we are far behind/first time syncing to this peer
in both cases we should validate the WP

Reply:
I removed the `syncing` check, but on second thought I'm not sure that's correct? We only want to skip the weight proof validation if we're going down the secondary sync path (when `syncing` is true). If we're going to do a "primary" sync, we don't want to skip it.

Reply:
Also I don’t think your description of `syncing` is correct either.. it’s true if we are far behind or we have no synced peers. My understanding is that we want to do a quick sync from the fork point with the first peer and then kick off a sync from 0 in the background for subsequent peers while we continue handling peaks. Regardless, the naming is pretty confusing and should probably be called something like `sync_from_fork_point`.

Reply:
we always want to validate a weight proof prior to syncing to a peer 
if we are far behind or this is the first time syncing to this peer. i agree the naming here needs some work

Reply:
Right, and we always will, even with the `syncing` condition. The purpose of this check is simply to avoid the case where we validate a weight proof and then return without syncing because we are already performing a secondary sync. We need the `syncing` condition here to prevent the edge case where there is a secondary sync running but we are going down the primary sync path for this peer (probably rare but I think it's still worth guarding against).

Reply:
No worries, appreciate you taking a look! I've gone ahead and added back the syncing check. I believe the changes are safe now.

I was actually thinking of moving the weight proof request for secondary syncs into the background task so we can continue handling messages while fetching and validating the proof. I think this would fix the race condition you are referring to and also prevent a potential stall of up to six minutes (the default wp request timeout). But I decided against it because it would be a bigger refactor and I wasn't sure if there would be side effects. Something to consider for the future maybe.

Reply:
i was looking at this code some more and since we actually only call new_peak_wallet in _process_new_subscriptions and that allows only one concurrent call to new_peak_wallet, its looks like your suggestion will do the trick.

im working on refactoring new_peak_wallet, ill include your change

Reply:
Awesome, looking forward to seeing the refactor